### ISSUE TITLE
Enhance Layout/EmptyLineBetweenDefs to include macros

### ISSUE THREAD

**Issue by catwomey** [rubocop/rubocop#11738]
## Is your feature request related to a problem? Please describe.

When writing code, I utilize macros that accept blocks and behave like a method def. An example is the [rails test macro](https://github.com/rails/rails/blob/8030cff808657faa44828de001cd3b80364597de/activesupport/lib/active_support/testing/declarative.rb#L13).  Because these macros look like method definitions, I'd like them to follow the rules set in [Layout/EmptyLineBetweenDefs](https://docs.rubocop.org/rubocop/1.46/cops_layout.html#layoutemptylinebetweendefs).

## Describe the solution you'd like

Enhance Layout/EmptyLineBetweenDefs to format configured macros as if they were method definitions. Default configuration would be an empty list

This is how I expect it to work if I were to configure the rule with `['test']`

```ruby
# Bad
test "the truth" do
  assert true
end
test "the false" do
  assert_not false
end

# Good
test "the truth" do
  assert true
end

test "the false" do
  assert_not false
end
```

## Describe alternatives you've considered

I considered writing something specific to the test macro in [Rubocop-rails](https://github.com/rubocop/rubocop-rails). Similar to how [rubocop-rspec](https://docs.rubocop.org/rubocop-rspec/2.0/cops_rspec.html#rspecemptylineafterexamplegroup) is defined.  I wanted to see if there was appetite for something a little more generic in the core rubocop rules first.

## Additional context

I have some code written [here](https://github.com/rubocop/rubocop/compare/master...catwomey:rubocop:macro-lines-between) that I've tested against one of my repositories, and if there's interest from the maintainers here I'll open a pull request.

**Comment by nobuyo** [comment]
You might want to open a PR. I'm just a contributor, not a maintainer. But your work looks good to me.
The maintainers most likely give some feedbacks.
